client/signup/steps/website-content/index.tsx

```diff
@@ -171,6 +186,110 @@ function WebsiteContentStep( {
 	);
 }
 
+/**
+ * Hook that polls the /rest/v1.2/sites/<site fragment> API.
+ * It checks the site details for a valid DIFM purchase.
+ * A valid DIFM purchase has `.options.is_difm_lite_in_progress` set to `false`/`null`
+ * and `.options.difm_lite_site_options.pages` should be an array of strings.
+ *
+ * If the above conditions are met, the hook returns { isLoading: false, hasValidPurchase: true }.
+ * If the above conditions are not met, it retries the request MAXTRIES times,
+ * with a linear backoff. If the conditions are still not met, the hook returns
+ * { isLoading: false, hasValidPurchase: false }.
+ * The default return value is { isLoading: true, hasValidPurchase: false }
+ *
+ * @param {(SiteId | null)} siteId The current site ID.
+ * @returns {{
+ * 	isLoading: boolean;
+ * 	hasValidPurchase: boolean;
+ * }}
+ */
+function usePollSiteForDIFMDetails( siteId: SiteId | null ): {
+	isLoading: boolean;
```

Comment:
Is Loading is a bit vague. Maybe something like isPollingComplete, or isPollingInProgress, or isAllRetriesCompleted.

And also maybe we can get the retry count as a parameter? So at a glance it's clear on the react component how many retries are being used.

Reply:
> Is Loading is a bit vague.

Good suggestion! 

> And also maybe we can get the retry count as a parameter? 

We can easily return `retryCount` here, but I'm wondering how this will be used in the component itself?

Reply:
Sorry I was not clear I mean define retryCount as an input parameter. So from the outer component level it can be more readable.
This is because when a function says they are polling the first question I get in my head is so Are we going to do it indefinitely?

Reply:
Ok, added!